### PR TITLE
CMakeLists: use flags for Apple correctly

### PR DESCRIPTION
**Pull request by barracuda156**
Currently CMakeLists hardcode `libc++`, which is wrong and gonna break the build with GCC. Use the flag correctly, only with Clang.

### CODE REVIEW

**Comment by barracuda156** [comment]
P. S. The following commit has introduced broken code: https://github.com/LMMS/lmms/commit/59276a03075055581dad0b8bf70bbb6ced0182c0

File: plugins/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ SET(CMAKE_DEBUG_POSTFIX "")
 # Enable C++17
 SET(CMAKE_CXX_STANDARD 17)
 
-IF(LMMS_BUILD_APPLE)
+IF(LMMS_BUILD_APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (tresf) [inline plugins/CMakeLists.txt:8]: Hmm.... I think it's `AppleClang`, e.g:

https://github.com/LMMS/lmms/blob/c53f3d981b9710bb1601e9213f925d95bdd9bd72/cmake/modules/CompileCache.cmake#L11

Per: https://cmake.org/cmake/help/latest/policy/CMP0025.html

**Comment by tresf** [comment]
> P. S. The following commit has introduced broken code: [59276a0](https://github.com/LMMS/lmms/commit/59276a03075055581dad0b8bf70bbb6ced0182c0)

If it's isolated to these two lines, removing the variable is probably justified.  If it's needed in more places, we can revert that commit.

Somewhat related.... Do you know how Clang vs. AppleClang behaves?

File: plugins/CMakeLists.txt (hunk 1 above)
Comment (barracuda156) [reply plugins/CMakeLists.txt:8]: `MATCHES` covers that (`STREQUAL` would not). Earlier CMake policy will use Clang for Apple Clang, so it is not sure we have AppleClang as an ID. Unless the policy is enforced here, and then either is fine.

**Comment by barracuda156** [comment]
@tresf 
> If it's isolated to these two lines, removing the variable is probably justified. If it's needed in more places, we can revert that commit.

I do not know what was the reason behind that change (there might have been some, maybe someone was building with LLVM Clang, but GNU `libstdc++`, which would be weird, but who knows). So I am not asking to revert it – I cannot guarantee is won’t have some effects for someone else. However as-is, the code breaks GCC build, so it should be fixed.
(Normally compiler itself adds the correct runtime flags. At least GCC does – I think we never need `-libstdc++` to be passed manually.)

> Do you know how Clang vs. AppleClang behaves?

I am perhaps not the right person to ask about Clang specifics, but from experience in Macports, in most cases they are interchangeable. I think LLVM one is more robust – when Xcode one fails to build something, switching to LLVM one works.

Or if you refer to CMakeLists, then using MATCHES will work for both (this is sure and tested). On the other hand, if one Clang should be included but another excluded, then STREQUAL with the specific compiler ID is likely the right thing.

**Comment by tresf** [comment]
> I think we never need -libstdc++ to be passed manually.)

I added this because there was a cutover period when the compiler did not assume this flag.  It's probably best to keep it if we have active back porters (or at least document and add a conditional for which clang versions may assume the wrong one before we remove it).

I can probably track this requirement down if needed, but it would be digging up some old PRs, probably for 10.7 or 10.8 support, before Homebrew took away support.

**Comment by tresf** [comment]
Found it... https://github.com/LMMS/lmms/commit/2035ff3e3c0d5cef7882dc88e4b4e4fb86da8335

Lion needed it.

**Comment by tresf** [comment]
The bug report that inspired it: https://github.com/LMMS/lmms/issues/2263

**Comment by barracuda156** [comment]
> Found it... [2035ff3](https://github.com/LMMS/lmms/commit/2035ff3e3c0d5cef7882dc88e4b4e4fb86da8335)
> 
> Lion needed it.

That one was still okay, since it used the flag only for Clang. But anyway, now both Clang and GCC should be happy.

P. S. Thank you very much for merging!

**Comment by tresf** [comment]
@barracuda156 does this need to be backported to 1.2 for your build to succeed?

**Comment by barracuda156** [comment]
> @barracuda156 does this need to be backported to 1.2 for your build to succeed?

This will be needed for any version, as long as GCC is used on Apple (regardless of the build arch). If it can be backported to 1.2.2, it will be nice for sure.

**Comment by tresf** [comment]
> > @barracuda156 does this need to be backported to 1.2 for your build to succeed?
> 
> This will be needed for any version, as long as GCC is used on Apple (regardless of the build arch). If it can be backported to 1.2.2, it will be nice for sure.
> 
> P. S. If 1.2.2 gets updated with backport of fixes, we need this as well: [ae4d1b5](https://github.com/LMMS/lmms/commit/ae4d1b52e2b4f871a006a777d8dc4a2bae688329)

Backported to `stable-1.2` via afaceadce.

**Comment by tresf** [comment]
> P. S. If 1.2.2 gets updated with backport of fixes, we need this as well: [ae4d1b5](https://github.com/LMMS/lmms/commit/ae4d1b52e2b4f871a006a777d8dc4a2bae688329)

Hi, since this won't cherry-pick cleanly, I'll ask either you or the OP of ae4d1b5 to create a separate PR against `stable-1.2` with these changes.  You may use `git cherry-pick ...` and resolve conflicts or -- if you wish -- you may create a fresh commit with these changes, we're happy either way.

**Comment by barracuda156** [comment]
@tresf Thank you!

P. S. _Possibly_ we do not need RISC-V in 1.2.x, I am not sure it is actually supported there otherwise.

**Comment by tresf** [comment]
> @tresf Thank you!
> 
> P. S. _Possibly_ we do not need RISC-V in 1.2.x, I am not sure it is actually supported there otherwise.

Correct.

**Pull request merged**